src/cdk/a11y/live-announcer/live-announcer.ts

```diff
@@ -149,17 +160,57 @@ export class LiveAnnouncer implements OnDestroy {
     }
   }
 
-  private _createLiveElement(): HTMLElement {
-    const elementClass = 'cdk-live-announcer-element';
-    const previousElements = this._document.getElementsByClassName(elementClass);
-    const liveEl = this._document.createElement('div');
+  private _enqueueAnnouncement(
+      message: string, politeness: AriaLivePoliteness, duration?: number): Promise<void> {
+    const liveElement = this._createLiveElement();
+    // TODO: ensure changing the politeness works on all environments we support.
+    liveElement.setAttribute('aria-live', politeness);
+
+    return new Promise(resolve => {
+      this._taskQueue.enqueue(() => {
+        liveElement.textContent = message;
+        resolve();
+
+        duration = duration ? duration : ARIA_LIVE_ANNOUNCER_REMOVAL_DELAY_MS;
+        setTimeout(() => {
+          document.body.removeChild(liveElement);
+        }, duration);
+      }, politeness, ARIA_LIVE_ANNOUNCER_DELAY_MS);
+    });
+  }
+
+  private _announceImmediately(
+      message: string, politeness: AriaLivePoliteness, duration?: number): Promise<void> {
+    // TODO: ensure changing the politeness works on all environments we support.
+    this._liveElement.setAttribute('aria-live', politeness);
+
+    return this._ngZone.runOutsideAngular(() => {
+      return new Promise(resolve => {
+        clearTimeout(this._previousTimeout);
+        this._previousTimeout = setTimeout(() => {
```

Comment:
Should run outside the `NgZone`.

Reply:
I'm unfamiliar with how zone.js works. The promise is already wrapped in `ngZone.runOutsideAngular`, is it enough or there should be another similar wrapper around `setTimeout`?

Also this particular code was just moved from another location. If it's a bug, let me know, it's probably better to fix it in a separate PR